crates/subspace-node/src/commands/run.rs

```diff
@@ -372,3 +380,50 @@ pub async fn run(run_options: RunOptions) -> Result<(), Error> {
         .await
         .map_err(Into::into)
 }
+
+pub fn ensure_block_and_state_pruning_params(config: &mut Configuration) {
+    let blocks_to_prune =
+        DOMAINS_BLOCK_PRUNING_DEPTH.saturating_mul(DOMAINS_PRUNING_DEPTH_MULTIPLIER);
+
+    if let BlocksPruning::Some(blocks) = config.blocks_pruning {
+        config.blocks_pruning = BlocksPruning::Some(if blocks >= blocks_to_prune {
+            blocks
+        } else {
+            warn!(
+                "Blocks pruning constraints needs to be atleast {:?}",
+                blocks_to_prune
+            );
+            blocks_to_prune
+        });
+    }
+
+    match &config.state_pruning {
+        None => {
+            config.state_pruning = Some(PruningMode::Constrained(Constraints {
+                max_blocks: Some(blocks_to_prune),
+            }))
+        }
+        Some(pruning_mode) => {
+            if let PruningMode::Constrained(constraints) = pruning_mode {
+                let blocks_to_prune = match constraints.max_blocks {
+                    None => blocks_to_prune,
+                    Some(blocks) => {
+                        if blocks >= blocks_to_prune {
+                            blocks
+                        } else {
+                            warn!(
+                                "State pruning constraints needs to be atleast {:?}",
```

Comment:
It might be helpful to use the names of the enum variant and config here:
```suggestion
                                "State pruning constrained config needs to be at least {:?}",
```